CDP4Common/Helpers/NestedElementTreeGenerator.cs

```diff
@@ -94,6 +94,45 @@ public IEnumerable<NestedElement> Generate(Option option, DomainOfExpertise doma
             return createNestedElements;
         }
 
+        /// <summary>
+        /// Returns the <see cref="NestedParameter"/>s in a flat list
+        /// </summary>
+        /// <param name="option">
+        /// The <see cref="Option"/> for which the <see cref="NestedElement"/> tree is created. When the <see cref="Option"/>
+        /// is null then none of the <see cref="ElementUsage"/>s are filtered.
+        /// </param>
+        /// <param name="domainOfExpertise">
+        /// The <see cref="DomainOfExpertise"/> for which the <see cref="NestedElement"/> tree needs to be generated. Only the <see cref="Parameter"/>s, <see cref="ParameterOverride"/>s and
+        /// <see cref="ParameterSubscription"/>s that are owned by the <see cref="DomainOfExpertise"/> will be taken into account when generating <see cref="NestedParameter"/>s
+        /// </param>
+        /// <returns>
+        /// An <see cref="IEnumerable{NestedElement}"/> that contains the generated <see cref="NestedElement"/>s
+        /// </returns>
+        /// <exception cref="ArgumentNullException">
+        /// thrown when the <paramref name="domainOfExpertise"/> is null
+        /// thrown when the <paramref name="option"/> is null
+        /// </exception>
+        public IEnumerable<NestedParameter> GetNestedParameters(Option option, DomainOfExpertise domainOfExpertise)
+        {
+            if (option == null)
+            {
+                throw new ArgumentNullException("option", "The option may not be null");
+            }
+
+            if (domainOfExpertise == null)
+            {
+                throw new ArgumentNullException("domainOfExpertise", "The domainOfExpertise may not be null");
+            }
+
+            var iteration = (Iteration)option.Container;
+
+            Logger.Debug($"Generating NestedParameter for Iteration {iteration.Iid}, Option: {option.ShortName}, DomainOfExpertise {domainOfExpertise.ShortName}");
+

```

Comment:
The code at line 131 does not take into account that `ElementUsages` may be excluded from the option and should therefore be omitted. Also, the root node is omitted as well. This needs some improvements.